tests/chainer_tests/links_tests/connection_tests/test_convolution_2d.py

```diff
@@ -305,17 +304,50 @@ def test_forward(self, backend_config):
             (self.height - self.kernel_height + 1) // self.strides_height,
             (self.width - self.kernel_width + 1) // self.strides_width)
 
-    def test_forward_invalid_backend(self):
+
+@testing.inject_backend_tests(
+    [
+        'test_forward',
+    ],
+    # CPU tests
+    [{}]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never'],
+    }))
+class TestConvolution2DMemoryLayoutsInvalid(unittest.TestCase,
+                                            TestConvolution2DLayoutParameters):
```

Comment:
Please derive from a common base class, instead of another concrete class.
It's confusing to overwriting an existing test implementation, and it's the convention of existing tests to avoid duplication.

Reply:
Do you mean to inherit from only one class?

Reply:
Yes.
```
Convolution2DMemoryLayoutsTestBase
         ^         ^
         |         +--- TestConvolution2DMemoryLayouts
         +------  TestConvolution2DMemoryLayoutsInvalid
```

Reply:
Oh, both concrete classes should inherit from `unittest.TestCase` too.